lib/nerves_hub_web/channels/device_channel.ex

```diff
@@ -49,8 +49,11 @@ defmodule NervesHubWeb.DeviceChannel do
 
     send(self(), :device_registation)
 
-    # Request device extension capabilities
-    push(socket, "extensions:get", %{})
+    # Request device extension capabilities if possible
+    # Earlier versions of nerves_hub_link don't have a fallback for unknown messages,
+    # so check version before requesting extensions
+    if is_safe_to_request_extensions?(params["device_api_version"]),
+      do: push(socket, "extensions:get", %{})
```

Comment:
```suggestion
    if is_safe_to_request_extensions?(socket.assigns.device_api_version),
      do: push(socket, "extensions:get", %{})
```
If we move this to after all the assigns, we can piggy back on the safe default that `assign_api_version` gives us

Reply:
Gosh dangit lol Thanks for the callout.